Kludex/starlette
# Version 1.0.0

- Closes #51

## Description

This PR bumps Starlette to 1.0.0. :tada:

## Changes

- Bump Starlette to 1.0.0.
- Add deprecation policy (`docs/deprecation.md`).
- Remove all code that was marked as deprecated.
- ~Deprecate `BaseHTTPMiddleware`.~
- ~Deprecate decorator functions in the `Router` method, which were discouraged for some time.~

> **Note**
> UPDATE [05/10/22]: PRs created according to comments on this PR:
> - https://github.com/encode/starlette/pull/1898 
> - https://github.com/encode/starlette/pull/1897

## Blockers

A blocker is PR or idea that can potentially make us jump to 2.0.0 too fast. If you see one around, please comment.

- [x] Decision on `UploadFile` PRs:
	- [x] https://github.com/encode/starlette/pull/1405
	- [x] https://github.com/encode/starlette/pull/1413
- [x] Decision on `Request.url_for` or related PRs:
	- [x] https://github.com/encode/starlette/pull/1385
	- [x] https://github.com/encode/starlette/pull/611
	- [x] https://github.com/encode/starlette/pull/1738 - No need to be included.
	- [x] https://github.com/encode/starlette/pull/1416 - No need to be included.
- [x] **Important**: check if #1858 is really an issue.  - No need to be included.

> **Note**
> UPDATE [22/10/22]: We need to fix the `python-multpart` issues.
> - [x] #1059

## Missing

- [X] Deprecate decorators on `Starlette` - I've only deprecated on `Router` until now.

> **Note**
> UPDATE [05/10/22]: Based on the comments on this PR, there's an agreement that Starlette 1.0 will not have deprecation warnings. We shall remove all deprecated code before that.

## Questions

1. Should we remove the code around the `lifespan` warnings? Is that part of the code supposed to be removed?
2. Should we remove the `allow_redirects` deprecation? It was recently added.
    - Yes. It's been months already.
4. Should we think about release cycle?
    - We are going to support only the latest version.

## How to review this PR?

1. Check `warnings.warn` in code.
2. Read the `Deprecation Policy` section using `./scripts/docs serve`.
3. Try hyperlinks used in the code, and release notes.
